server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -109,6 +112,11 @@ public class TransportBulkAction extends HandledTransportAction<BulkRequest, Bul
     private static final String DROPPED_ITEM_WITH_AUTO_GENERATED_ID = "auto-generated";
     private final IndexingPressure indexingPressure;
     private final SystemIndices systemIndices;
+    private final InferenceProvider inferenceProvider;
+
+    public static final String ROOT_RESULT_FIELD = "_ml_inference";
+    public static final String INFERENCE_FIELD = "result";
+    public static final String TEXT_FIELD = "text";
```

Comment:
We need to share the root field options that contain the results. Right now https://github.com/elastic/elasticsearch/pull/103978 has it as `_semantic_text_inference` and the embedding itself is called `sparse_embedding`.

This makes me think that either:

 - The field mapper should indicate the metadata-fields it requires and we use those
 - Inference provider knows how to map the results correctly and the bulk action just accepts those.